tests/dataset_builders/hf/drugprot/test_hf_drugprot.py

```diff
@@ -0,0 +1,293 @@
+                    "type": "GENE-N",
+                    "text": ["RDHs"],
+                    "offsets": [[139, 143]],
+                    "normalized": [],
+                },
+                {
+                    "id": "17512723_T9",
+                    "type": "GENE-Y",
+                    "text": ["human type 12 RDH"],
+                    "offsets": [[417, 434]],
+                    "normalized": [],
+                },
+                {
+                    "id": "17512723_T10",
+                    "type": "GENE-N",
+                    "text": ["retinol dehydrogenases"],
+                    "offsets": [[115, 137]],
+                    "normalized": [],
+                },
+                {
+                    "id": "17512723_T11",
+                    "type": "GENE-N",
+                    "text": ["human and murine RDH 12"],
+                    "offsets": [[191, 214]],
+                    "normalized": [],
+                },
+                {
+                    "id": "17512723_T12",
+                    "type": "GENE-Y",
+                    "text": ["RDH12"],
+                    "offsets": [[0, 5]],
+                    "normalized": [],
+                },
+                {
+                    "id": "17512723_T13",
+                    "type": "GENE-N",
+                    "text": ["retinol dehydrogenase"],
+                    "offsets": [[9, 30]],
+                    "normalized": [],
+                },
+            ],
+            "events": [],
+            "coreferences": [],
+            "relations": [
+                {
+                    "id": "17512723_0",
+                    "type": "PRODUCT-OF",
+                    "arg1_id": "17512723_T1",
+                    "arg2_id": "17512723_T9",
+                    "normalized": [],
+                }
+            ],
+        }
+    else:
+        raise ValueError(f"Unknown dataset variant: {dataset_variant}")
+
+
+def test_hf_example_for_every_split(hf_dataset, dataset_variant, split):
+    # covers both dataset variants
+    example = hf_dataset[split][0]
+    if split == "train" and dataset_variant == "drugprot_bigbio_kb":
+        assert example["document_id"] == "17512723"
+        assert len(example["entities"]) == 13
+        assert len(example["relations"]) == 1
+    elif split == "validation":
+        assert example["document_id"] == "17651117"
+        assert len(example["entities"]) == 18
+        assert len(example["relations"]) == 0
+    elif split == "test_background":
+        assert example["document_id"] == "32733640"
+        assert len(example["entities"]) == 37
+        assert len(example["relations"]) == 0
+
+
+def test_hf_dataset_all(hf_dataset, split):
+    # covers both dataset variants
+    for example in hf_dataset[split]:
+        assert example["document_id"] is not None
+        assert example["entities"] is not None
+        if split != "test_background":
+            assert example["relations"] is not None
```

Comment:
what do you want to check here? this subsumes below condition (`assert example["relations"] == []`) ...

Reply:
better:

```python
assert example["relations"] is not None
if  split == "test_background":
     assert example["relations"] == []
```